server/hotrod/src/main/java/org/infinispan/server/hotrod/tx/table/GlobalTxTable.java

```diff
@@ -257,8 +254,9 @@ private void onOngoingTransaction(CacheXid cacheXid, TxState state) {
             //local transaction doesn't exists.
             onTransactionCompleted(cacheXid);
          } else {
-            blockingExecutor.execute(
-                  () -> rollbackOldTransaction(cacheXid, state, () -> completeLocal(txTable, cacheXid, tx, false)));
+            blockingManager.runBlocking(
+                  () -> rollbackOldTransaction(cacheXid, state, () -> completeLocal(txTable, cacheXid, tx, false)),
```

Comment:
IMO `rollbackOldTransaction` should return `CompletionStage`

Reply:
Hrmm... yeah interesting. This will actually change the behavior to wait for the rollback to complete. However I noticed that line 248 also doesn't wait. I think I will create a JIRA for someone to look into this separately, since I am not familiar with it and don't want it to hold this up.

https://issues.redhat.com/browse/ISPN-11722

Reply:
Ok